iroh-store/src/store.rs

```diff
@@ -33,7 +28,7 @@ pub struct Store {
 
 struct InnerStore {
     content: RocksDb,
-    next_id: AtomicU64,
+    next_id: RwLock<u64>,
```

Comment:
The problem is that the LocalStore is not Send, since some internals of RocksDB are not Send, so the column family handles are not send. I briefly tried tokio mutexes, but that would have required more changes. Using a parking lot mutex is the least invasive way to do this.

It is a reader/writer lock, so you can read from multiple threads usually. If you have lots of concurrent writes you would have threads piling up, yes.

Note that the `spawn_blocking` is probably the right thing to do in any case, because the rocksdb db ops are synchronous, and you are not supposed to do synchronous blocking IO on one of the tokio thread pool threads.

Reply:
oh, good poing on `spawn_blocking` beeing needed for the sync I/O anyway.  so yeah, then my comment doesn't make much sense.

Reply:
So, I might go with tokio mutexes eventually after all. Just don't want to do it right now, since it requires larger changes and I want to get a simple fix in first for the bug that has been plaguing us since before ipfs camp.